tests/v2/validation/provisioning/permutations/permutations.go

```diff
@@ -0,0 +1,171 @@
+		for _, kubeVersion := range kubeVersions {
+			for _, cni := range clustersConfig.Basics.CNIs {
+				testClusterConfig = ConvertConfigToClusterConfig(clustersConfig)
+				testClusterConfig.CNI = cni
+				name = testNamePrefix + " Node Provider: " + nodeProviderName + " Kubernetes version: " + kubeVersion + " cni: " + cni
+				s.Run(name, func() {
+					switch clusterType {
+					case RKE2ProvisionCluster, K3SProvisionCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						if hostnameTruncation == nil {
+							hostnameTruncation = &machinepools.HostnameTruncation{
+								PoolNameLengthLimit:    0,
+								ClusterNameLengthLimit: 0,
+							}
+						}
+						clusterObject, err := crud.CreateProvisioningCluster(client, *nodeProvider, testClusterConfig, hostnameTruncation)
+
+						require.NoError(s.T(), err)
+						verify.VerifyCluster(s.T(), client, clusterObject)
+
+					case RKE1ProvisionCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						nodeTemplate, err := rke1Provider.NodeTemplateFunc(client)
+						require.NoError(s.T(), err)
+
+						clusterObject, err := crud.CreateProvisioningRKE1Cluster(client, *rke1Provider, testClusterConfig, nodeTemplate)
+						require.NoError(s.T(), err)
+
+						verify.VerifyRKE1Cluster(s.T(), client, testClusterConfig, clusterObject)
+
+					case RKE2CustomCluster, K3SCustomCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+
+						clusterObject, err := crud.CreateProvisioningCustomCluster(client, *customProvider, testClusterConfig)
+						require.NoError(s.T(), err)
+
+						verify.VerifyCluster(s.T(), client, clusterObject)
+
+					case RKE1CustomCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						clusterObject, err := crud.CreateProvisioningRKE1CustomCluster(client, *customProvider, testClusterConfig)
+						require.NoError(s.T(), err)
+
+						verify.VerifyRKE1Cluster(s.T(), client, testClusterConfig, clusterObject)
+
+					// airgap currently uses corral to create nodes and register with rancher
+					case RKE2AirgapCluster, K3SAirgapCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						clusterObject, err := crud.CreateProvisioningAirgapCustomCluster(client, testClusterConfig, corralPackages)
+						require.NoError(s.T(), err)
+
+						verify.VerifyCluster(s.T(), client, clusterObject)
+
+					case RKE1AirgapCluster:
+						testClusterConfig.KubernetesVersion = kubeVersion
+						clusterObject, err := crud.CreateProvisioningRKE1AirgapCustomCluster(client, testClusterConfig, corralPackages)
+						require.NoError(s.T(), err)
+
+						verify.VerifyRKE1Cluster(s.T(), client, testClusterConfig, clusterObject)
+
+					default:
+						s.T().Fatalf("Invalid cluster type: %s", clusterType)
+					}
+				})
+			}
+		}
+	}
+}
+
+func ConvertConfigToClusterConfig(clustersConfig *provisioning.Config) *provisioning.ClusterConfig {
```

Comment:
Let's move this function into extensions/provisioning since this func will be used outside of provisioning